### PR TITLE
Users should be directly able to navigate to the product detail page (#1yzcv99).

### PR DESCRIPTION
**Pull request by meet-aniket**
image

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const state: any = {
 }
 
 const persistState = createPersistedState({
-    paths: ['user', 'product.uploadProducts'],
+    paths: ['user', 'product.uploadProducts', 'product.current'],
```
[the remark this answers is not in the document]
Comment (meet-aniket) [reply src/store/index.ts:21]: Removed current from persist state and tested whether we are losing or not our current product.

File: src/store/modules/product/actions.ts [hunk 2]
```diff
@@ -83,10 +83,19 @@ const actions: ActionTree<ProductState, RootState> = {
...
-  updateCurrentProduct ({ commit, state }, payload) {
+  async updateCurrentProduct ({ commit, state }, payload) {
     // search in uploadProducts that if the clicked product is already in the upload list and set it as current product
-    const currentProduct = state.uploadProducts[payload.product.sku]
-    commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct ? currentProduct : payload.product })
+    const currentProduct = state.uploadProducts[payload.currentSku]
+    if(currentProduct){
+      commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct })
+    }
+    else{
```
[the remark this answers is not in the document]
Comment (meet-aniket) [reply src/store/modules/product/actions.ts]: Updated code and changed spacing around if-else.

File: src/store/modules/product/actions.ts [hunk 3]
```diff
@@ -83,10 +83,18 @@ const actions: ActionTree<ProductState, RootState> = {
...
-  updateCurrentProduct ({ commit, state }, payload) {
+  async updateCurrentProduct ({ commit, state }, payload) {
     // search in uploadProducts that if the clicked product is already in the upload list and set it as current product
-    const currentProduct = state.uploadProducts[payload.product.sku]
-    commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct ? currentProduct : payload.product })
+    const currentProduct = state.uploadProducts[payload]
+    if(currentProduct) {
+      commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct })
+    } else {
+      const resp = await ProductService.fetchProducts({
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:92]: Handle the case when API returns an error.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 3 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts:92]: Sir, updated code.

File: src/store/modules/product/actions.ts [hunk 4]
```diff
@@ -83,10 +83,22 @@ const actions: ActionTree<ProductState, RootState> = {
...
+        // used sku as we are currently only using sku to search for the product
+        "filters": ['sku: ' + '*' + payload + '*', 'isVirtual: false'],
+      })
+      if(resp.status === 200 && resp.data.response?.numFound > 0 && !hasError(resp)) {
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:96]:
```suggestion
      if(resp.status === 200 && !hasError(resp) && resp.data.response?.numFound > 0 ) {
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Pull request merged**